crates/wasi-http/wit/deps/filesystem/types.wit

```diff
@@ -538,8 +515,6 @@ interface types {
             open-flags: open-flags,
             /// Flags to use for the resulting descriptor.
             %flags: descriptor-flags,
-            /// Permissions to use when creating a new file.
-            modes: modes
```

Comment:
Question on this, if we know we want to add this in the future should we be trying to future-proof this function? For example we could add an `options: option<open-options>` with a currently empty `resource open-options` or something like that.

Alternatively we could add a feature to the component model and WIT for something like keyword arguments with defaults. That's a much longer- term vision though.

Reply:
I believe that we can eliminate the `modes` parameter here and let open-at continue to do the reasonable default behavior, and then when users of the re-introduced modes stuff want to opt into a non-default behavior for modes, we can expose a new `open-at-modes` function that has the additional argument. Then when it comes time to make a breaking change in preview 3 we can consolidate down to just one `open-at` with mode arguments.

We could get fancier than that, but I expect the above will work fine without introducing much complexity.